### PR TITLE
[R4R] fix: add ProxyOwnerUpgradeTime to mantle upgrade config

### PR DESCRIPTION
**Pull request by PinelliaC**
Change the proxyAdmin owner address to `0x09734bB3980906Bb217305EA6Bd34256feEAB105`.

### CODE REVIEW

File: core/state_transition.go [hunk 1]
```diff
@@ -467,6 +467,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	snap := st.state.Snapshot()
 	// Will be reverted if failed
 
+	// Check if the owner of the L2ProxyAdmin contract needs to be upgraded
+	if st.evm.ChainConfig().IsProxyOwnerUpgrade(&st.evm.Context.Time) {
```
Comment (pandainzoo) [inline core/state_transition.go:471]: need check chainID here

File: core/mantle_upgrade.go [hunk 2]
```diff
@@ -13,50 +14,72 @@ var (
...
+
+	// OwnerSlot refers to the storage slot in the L2ProxyAdmin contract that
+	// holds the owner of the contract
+	OwnerSlot = common.BigToHash(big.NewInt(0))
+
+	// NewProxyAdminOwnerAddress is the address that the L2ProxyAdmin contract
+	// will be transferred to
+	NewProxyAdminOwnerAddress = common.HexToHash("0x09734bB3980906Bb217305EA6Bd34256feEAB105")
+)
+
```
Comment (pandainzoo) [inline core/mantle_upgrade.go:74]: it's better to mv this to state_transition.go, only the upgrade height is stored here

File: core/mantle_upgrade.go [hunk 3]
```diff
@@ -13,50 +14,72 @@ var (
...
 		MetaTxV3UpgradeTime:   nil, //TODO set upgrade timestamp
+		ProxyOwnerUpgradeTime: nil, //TODO set upgrade timestamp
 	}
 	MantleSepoliaUpgradeConfig = MantleUpgradeChainConfig{
 		ChainID:               params.MantleSepoliaChainId,
 		BaseFeeTime:           u64Ptr(1_704_891_600),
 		BVMETHMintUpgradeTime: u64Ptr(1_720_594_800),
 		MetaTxV2UpgradeTime:   u64Ptr(1_720_594_800),
 		MetaTxV3UpgradeTime:   u64Ptr(1_720_594_800),
+		ProxyOwnerUpgradeTime: u64Ptr(0),
```
Comment (pandainzoo) [inline core/mantle_upgrade.go:25]: this should be nil

File: core/mantle_upgrade.go (hunk 2 above)
Comment (PinelliaC) [reply core/mantle_upgrade.go]: fixed

File: core/mantle_upgrade.go (hunk 3 above)
Comment (PinelliaC) [reply core/mantle_upgrade.go]: fixed

File: core/state_transition.go (hunk 1 above)
Comment (PinelliaC) [reply core/state_transition.go]: Separated according to chainconfig

**Review by pandainzoo — APPROVED** [review]

**Pull request merged**